src/js/events/MesosSummaryActions.js

```diff
@@ -1,58 +1,95 @@
-import ActionTypes from '../constants/ActionTypes';
+import {
+  REQUEST_MESOS_HISTORY_SUCCESS,
+  REQUEST_MESOS_HISTORY_ONGOING,
+  REQUEST_MESOS_SUMMARY_SUCCESS,
+  REQUEST_MESOS_SUMMARY_ERROR,
+  REQUEST_MESOS_SUMMARY_ONGOING
+} from '../constants/ActionTypes';
+
 var AppDispatcher = require('./AppDispatcher');
 var Config = require('../config/Config');
 var RequestUtil = require('../utils/RequestUtil');
 var _historyServiceOnline = true;
 
-function getStateUrl(timeScale) {
-  timeScale = timeScale || 'last';
-  if (_historyServiceOnline) {
-    return `${Config.historyServer}/dcos-history-service/history/${timeScale}`;
-  } else {
-    return `${Config.rootUrl}/mesos/master/state-summary`;
+function testHistoryOnline() {
+  RequestUtil.json({
+    url: `${Config.historyServer}/dcos-history-service/history/last`,
+    success: function () {
+      _historyServiceOnline = true;
+    },
+    error: function () {
+      setTimeout(testHistoryOnline, Config.testHistoryInterval);
+    }
+  });
+}
+
+function getHistory(resolve, reject, timeScale = 'last') {
+  let url = `${Config.historyServer}/dcos-history-service/history/${timeScale}`;
+  let successEventType = REQUEST_MESOS_SUMMARY_SUCCESS;
+
+  if (timeScale === 'minute') {
+    successEventType = REQUEST_MESOS_HISTORY_SUCCESS;
   }
+
+  RequestUtil.json({
+    url,
+    success: function (response) {
+      AppDispatcher.handleServerAction({
+        type: successEventType,
+        data: response
+      });
+      resolve();
+    },
+    error: function () {
```

Comment:
Because we don't need it. We immediately fall back on `state-summary`. Previously we emitted the error and then waited for the next round of polling before we switched to `state-summary`. Now on error, we immediately delegate to `state-summary` and if that fails, we dispatch an error.